### ISSUE TITLE
[Bug]: Switched to wayland, apps look strange

### ISSUE THREAD

**Issue by tim-gromeyer** [flatpak/flatpak#5606]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a bug that matches the one I want to file, without success.
- [X] If this is an issue with a particular app, I have tried filing it in the appropriate issue tracker for the app (e.g. under https://github.com/flathub/) and determined that it is an issue with Flatpak itself.
- [X] This issue is not a report of a security vulnerability (see [here](https://github.com/flatpak/flatpak/blob/main/SECURITY.md) if you need to report a security issue).

### Flatpak version

Flatpak 1.14.4

### What Linux distribution are you using?

Ubuntu

### Linux distribution version

Kubuntu 23.10

### What architecture are you using?

x86_64

### How to reproduce

Use flatpak apps, switch to wayland

### Expected Behavior

Look like in X11

### Actual Behavior

Very large cursor, text looks strange
Wayland: 
![image](https://github.com/flatpak/flatpak/assets/58736434/d4fccd23-98ed-45c4-a728-46ffaf4376b2)

X11:
![image](https://github.com/flatpak/flatpak/assets/58736434/1d6cb1a3-6ddf-4566-81b1-647c27440aca)

I use the default cursor, have no flatpak overrides, and use kde. Everything worked fine in X11, but now the cursor is huge and the text looks ugly. Also, the cursor doesn't change. I go over the edge of the window with the mouse and the cursor gets huge and changes to indicate that I can resize the window, but once I'm inside the window. it doesn't change back as it should.

In Wayland 1 icon also differs for some reason, and the title bas decorations too (in wayland the maximize and minize buttons are missing for this specific app)

Demo video: [here](https://drive.google.com/file/d/12KdB40PbDXtVL0LWTYvv8RTPFGa8TeMX/view?usp=drive_link)

### Additional Information

This happens in every app I havem the are all GTK4 Libadwaida.

**Comment by JakobDev** [comment]
You should also report this on the [KDE Bugtracker](https://bugs.kde.org/)

**Comment by cchulo** [comment]
I am experiencing this same problem, bottles is completely unusable, are we certain this is KDE only problem? can anyone on Gnome confirm they're not seeing this issue?

**Comment by flexagoon** [comment]
@cchulo do you have fractional scaling enabled? It doesn't matter if you're actually using it or not, even just having the feature enabled changes some of the rendering mechanisms.

**Comment by cchulo** [comment]
> @cchulo do you have fractional scaling enabled? It doesn't matter if you're actually using it or not, even just having the feature enabled changes some of the rendering mechanisms.

i don't think so, scaling is currently set to 100%, and generally never mess with it as the defaults are fine for me at 1440p. Is that what youre talking about? to my knowledge, turning off fractional scaling is to keep scaling at increments of 100, so keeping it at 100% is technically turning it off, right?

**Comment by flexagoon** [comment]
> keeping it at 100% is technically turning it off, right?

@cchulo the feature is turned on if you have the option to set the scaling to a value which is not a multiple of 100. So if the "display settings" page allows you to set scaling to eg. 125%, that means the feature is enabled.

However, since you're having this issue with GTK apps, and scaling issues usually affect Electron apps, this may not be related to that. This is probably a duplicate of #2861.

**Comment by rezad1393** [comment]
is there any workaround outside of forcing them to be run as xwayland?

as the other user said many apps have this issue and for me bottles is shown really bad.

these are two screenshot of bottles with wayland used and not allowed via flatseal
![2024-01-26_14-47_1](https://github.com/flatpak/flatpak/assets/7523164/1c7938dd-5288-427b-9938-31bfa4d9dedb)
![2024-01-26_14-47](https://github.com/flatpak/flatpak/assets/7523164/69857c15-522b-4d42-9924-73de770a3934)

**Comment by rezad1393** [comment]
the workaround does what exactly?
I didn't have the xdg-desktop-portal-gtk installed and I didn't try before using your command so I don't know if your command worked or that package was needed.

is that workaround reversible?

**Comment by cchulo** [comment]
the user HBR here gave a brief description of what it does. I don't have time to nitpick or dissect exactly what its doing and will take the word of the expert in this case: https://forum.endeavouros.com/t/odd-flatpak-graphics-on-wayland/50394/2

**Comment by flexagoon** [comment]
@rezad1393 try installing xdg-desktop-portal-gtk

**Comment by rezad1393** [comment]
> the user HBR here gave a brief description of what it does. I don't have time to nitpick or dissect exactly what its doing and will take the word of the expert in this case: https://forum.endeavouros.com/t/odd-flatpak-graphics-on-wayland/50394/2

that one doesnt say 
"/usr/lib/xdg-desktop-portal --replace"
I did both so I am wondering if this second part was necessary.

**Comment by flexagoon** [comment]
@rezad1393 the --replace thing basically just restarts the running instance of xdg-desktop-portal, it's fine to run it

**Comment by rezad1393** [comment]
now I have issues with net.sourceforge.mcomix flatpak showing to icons on the left side.

**Comment by cchulo** [comment]
@rezad1393 i downloaded this flatpak and ran it, and you are right, the icons are missing. So its not a perfect fix i hate to say :(

**Comment by cchulo** [comment]
@rezad1393 quick follow up on your request for undoing my proposed fix: uninstall gtk portal, and return the `xdg-desktop-portal --replace` command again. I tested it myself

**Comment by cchulo** [comment]
@rezad1393 quick follow up on your request for undoing my proposed fix: uninstall gtk portal, and rerun the `/usr/lib/xdg-desktop-portal --replace` command again. I tested it myself, it works! sorry for the hassle, for me it works because i happen to use non-qt flatpaks for the most part

**Comment by chrisawi** [comment]
The non-antialiased text issue is the same as #2861, and that has been [fixed in GTK](https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/7090).

The skinny layout issue should also be fixed by that, as well as by a change in libadwaita to disregard a dpi of 0.

Other desktop integration issues like cursor theme and window controls can currently only be resolved by installing xdg-desktop-portal-gtk and configuring the [relevant GNOME settings](https://gitlab.gnome.org/GNOME/gtk/-/blob/4493c12801addb906f6092b924bafcde74d562a5/gdk/wayland/gdkdisplay-wayland.c?page=2#L1771-1810) with `gsettings`. There is no standardization between toolkits on Wayland for most settings like that.
